### ISSUE TITLE
Suggestion: shorter name for `setexcludinghandlers!`

### ISSUE THREAD

**Issue by BioTurboNick** [JuliaGizmos/Observables.jl#77]
`setexcludinghandlers!` is a bit long. Something like the following might be better:

`setsilent!`
`setquiet!`
`setmuted!`

And just a note here that if/when this is chosen, the examples in the GtkObservables.jl docs need to be updated. Currently uses `.value`, which doesn't work because that's not the current field name.

**Comment by hhaensel** [comment]
Hi @BioTurboNick , @SimonDanisch , @timholy 

I want to come back on my `[!]` solution that I proposed already in #67.
We use it in Stipple.jl for our own subtype `Reactive` and find it very useful.

```julia
o = Observable(1)

# standard assignment
o[] = 10

# silent assignment by an exclamation mark between the brackets
o[!] = 100
```

Furthermore we pass arguments that are not `!` on to the value of the observable and call notify in case of assignment.

```julia
o = [10, 20, 30]
o[2]
# 20

on(o) do x
    println("Hello $x")
end

o[2] = 200
# Hello [10, 200, 30]
```

The same with `getproperty`

```julia
mutable struct HH
    a::Integer
    b::Integer
end

o = Reactive(HH(1,2))

on(o) do x
    println("Hello $x")
end

o.a = 10
# Hello HH(10, 2)
```
Happy to hear what you think.

Note: Reactive has nothing to do with the package Reactive. It is an observable type `Reactive <: AbstractObservable`

**Comment by SimonDanisch** [comment]
This should really be a separate issue...
But, I also have a similar type in ShaderAbstractions:
https://github.com/SimonDanisch/ShaderAbstractions.jl/blob/master/src/types.jl#L12-L65=
Looking back on it, it doesn't seem that elegant, but maybe we can clean it up and add it to Observables.jl

**Comment by hhaensel** [comment]
We can make it a different issue with a better name.
Your idea is quite the same as mine.
If we 'd define these operations for AbstractObservables no macro would be needed

**Comment by SimonDanisch** [comment]
Yeah, that'd be an idea.
For my type it's important that one can register for certain updates so that one can do:
```julia
on(array, :setindex) do (idx, new_value)
...
end
```

**Comment by hhaensel** [comment]
Yeah, that's what we did. We declared Reactive <: AbstractObservable.
I'll submit a PR then :-)

**Comment by BioTurboNick** [comment]
Just to throw this out there - `_` might better indicate suppression than `!`

**Comment by treygreer** [comment]
Er, I wound up here trying to track down a bug in Interact.  It seems that WebIO throws an error looking for Observables.setexcludinghandlers!():  

line 346 in scope.jl, webIO v0.8.17:        Observables.setexcludinghandlers(ob, val, x -> !(x isa SyncCallback))

Complete newb here.  Any suggestions?
